File: 0_1.baekjoon/greedy/java/BOJ20365.java
import java.util.*;
import java.io.*;

public class BOJ20365 {
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int N = Integer.parseInt(br.readLine());
        String sequence = br.readLine();
        Map<Character, Integer> count = new HashMap<>();
        count.put('B', 0);
        count.put('R', 0);

        char current = sequence.charAt(0);
        char next = current;

        for (int i = 1; i < N; i++) {
            next = sequence.charAt(i);
            if (current != next) {
                count.compute(current, (key, value) -> value + 1);
                current = next;
            }
        }

        count.compute(next, (key, value) -> value + 1);

        if (sequence.length() == 1) {
            System.out.println(1);
        } else {
            System.out.println(Math.min(count.get('B'), count.get('R')) + 1);
        }       


    }
}

File: 0_1.baekjoon/greedy/kotlin/src/BOJ20365.kt
fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    val sequence = readLine()

    val count = mutableMapOf('R' to 0, 'B' to 0)
    var current = sequence[0]
    var next = current

    for (i in 1..n-1) {
        next = sequence[i]

        if (current != next) {
            count[current] = count[current]!! + 1
            current = next
        }
    }

    count[next] = count[next]!! + 1;

    if (sequence.length == 1) {
        println(1)
    } else {
        println(count.minOf { entry -> entry.value } + 1)
    }
}